index.rst

```diff
@@ -1,61 +1,311 @@
+
+Furthermore, through the ``visitId`` one can identify the source of the image (ComCam or LSSTCam), so there is no need for special commands for each one of them.
+
+At the same time, we can differentiate between an Intra/Extra wavefront analysis process and a corner wavefront sensors by the input alone.
+If only one input ``visitId`` then it can be assumed to be a corner wavefront sensor run.
+If, in addition to ``visitId`` an ``extraId`` is given, it can be considered and Intra/Extra wavefront process.
+
+The command would be used as follows:
+
+.. code-block:: py
+
+  from lsst.ts import salobj
+
+  d = salobj.Domain()
+
+  mtaos = salobj.Remote(d, "MTAOS")
+
+  await mtaos.start_task
+
+  # Process full array intra/extra focal images.
+  # Frames can be ComCam or LSSTCam
+  await mtaos.cmd_runWEP.set_start(
+      visitId=2020103000040, extraId=2020103000041
+  )
+
+  # Process corner wavefront sensor
+  # Frames should be LSSTCam only. If ComCam command will raise an exception.
+  await mtaos.cmd_runWEP.set_start(
+      visitId=2020103000040
+  )
+
+Source selection command
+------------------------
+
+In most cases, we will known the coordinates of the next visit field ahead of time.
+That means some pre-processing activities can be done in parallel, while the telescope slew and the image is acquired.
+One of these processes is to prepare a list of suitable targets for wavefront sensing.
+
+This could be done by adding a command that would cause the MTAOS to run the source selection routine.
+The command payload would minimally be;
+
+  - RA: Right ascension of the field (in hours).
+  - Dec: Declination of the field (in degrees).
+  - PA: Position angle (in degrees).
+  - mode: An enumeration specifying the wfs mode; corner, full array, etc.
+
+Issuing the command would look something like:
+
+.. code-block:: py
+
+  from lsst.ts.idl.MTAOS import WFSMode
+
+  # Select sources for Main Camera Corner wavefront sensor mode
+  await mtaos.cmd_selectSources.set_start(ra=14., dec=-10., pa=0., type=WFSMode.Corner)
+
+  # Select sources for Main Camera Full array mode
+  await mtaos.cmd_selectSources.set_start(ra=14., dec=-10., pa=0., type=WFSMode.FullArray)
+
+  # Select sources for ComCam mode
+  await mtaos.cmd_selectSources.set_start(ra=14., dec=-10., pa=0., type=WFSMode.ComCam)
+
+Pre-processing frames
+---------------------
+
+While obtaining Intra/Extra focal images for ComCam or Main Camera full array, one would have the first image ready while the second is exposing.
+That means, it would be possible to get ahead and start the pre-processing of those frames (e.g. ISR).
+
+This could be accomplished by adding a ``preProcess`` command to the MTAOS, which would take care of that initial step.
+The command would only need to know the ``dataId`` of the frame to process, e.g.;
```

Comment:
From the previous sections, I guess you are talking about the `visitId`. In addition, the `dataId` is a default argument of `butler.get()`. And the data type is `dict`.